src/state/products.js

```diff
@@ -58,15 +66,24 @@ export function isValidExtractedProduct(extractedProduct) {
     return false;
   }
 
-  for (const key of ['url', 'image']) {
-    try {
-      const url = new URL(extractedProduct[key]);
-      if (!['https:', 'http:'].includes(url.protocol)) {
-        return false;
-      }
-    } catch (err) {
+  try {
+    const url = new URL(extractedProduct.url);
+    const imageUrl = new URL(extractedProduct.image);
+
+    if (!['https:', 'http:'].includes(url.protocol)) {
+      return false;
+    }
+
+    // Some sites like Amazon initially load a data URI for the image
+    if (!['https:', 'http:', 'data:'].includes(imageUrl.protocol)) {
       return false;
     }
+
+    if (imageUrl.protocol === 'data:' && !(DATA_URI_PATHS.find(path => imageUrl.pathname.startsWith(path)))) {
```

Comment:
A small comment specifying that we're validating the mimetype of the image would help explain the line a bit more.